hammer_time/hammer_time.py

```diff
@@ -36,17 +37,33 @@ def get_auth_data(model_client):
     return cacert, username, password
 
 
+def remove_and_wait(client, machines):
+    conditions = []
+    for machine_id, data in machines:
+        client.juju('remove-machine', (machine_id,))
+        conditions.append(client.make_remove_machine_condition(machine_id))
+    client.wait_for(ConditionList(conditions))
```

Comment:
make_remove_machine_condition controls the timeout rules.  Most providers will time out after 10 minutes.  Azure will time out after 20 minutes.

This is why I wanted make_remove_machine_condition to be a member of ModelClient; so that it can tweak the timeouts based on provider.

When we implement ModelClient.remove_machine, I'd like it to return a WaitMachineNotPresent using the same logic.